File: src/main/java/io/github/chaosdave34/kitpvp/cosmetics/impl/SimpleKillEffect.java
package io.github.chaosdave34.kitpvp.cosmetics.impl;

import io.github.chaosdave34.kitpvp.cosmetics.KillEffect;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.Particle;

public class SimpleKillEffect extends KillEffect {
    private final Particle particle;

    public SimpleKillEffect(String id, String name, Particle particle, int levelRequirement, Material icon) {
        super(id, name, levelRequirement, icon);
        this.particle = particle;
    }

    @Override
    public void playEffect(Location location) {
        location.getWorld().spawnParticle(particle, location.add(0, 1, 0), 1, 0, 0, 0, 0);
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/cosmetics/impl/SimpleKillEffect.kt
package io.github.chaosdave34.kitpvp.cosmetics.impl

import io.github.chaosdave34.kitpvp.cosmetics.KillEffect
import org.bukkit.Location
import org.bukkit.Material
import org.bukkit.Particle

class SimpleKillEffect(id: String, name: String, val particle: Particle, levelRequirement: Int, icon: Material) : KillEffect(id, name, levelRequirement, icon) {
    override fun playEffect(location: Location) {
        location.world.spawnParticle(particle, location.add(0.0, 1.0, 0.0), 1, 0.0, 0.0, 0.0, 0.0)
    }
}